src/ShaderGen/Hlsl/HlslBackend.cs

```diff
@@ -183,47 +183,69 @@ HashSet<ResourceDefinition> resourcesUsed
 
                 foreach (ResourceDefinition rd in set)
                 {
-                    if (!resourcesUsed.Contains(rd))
-                    {
-                        continue;
-                    }
-
                     switch (rd.ResourceKind)
                     {
                         case ShaderResourceKind.Uniform:
-                            WriteUniform(sb, rd, uniformBinding);
+                            if (resourcesUsed.Contains(rd))
+                            {
+                                WriteUniform(sb, rd, uniformBinding);
+                            }
                             uniformBinding++;
                             break;
                         case ShaderResourceKind.Texture2D:
-                            WriteTexture2D(sb, rd, textureBinding);
+                            if (resourcesUsed.Contains(rd))
+                            {
+                                WriteTexture2D(sb, rd, textureBinding);
+                            }
                             textureBinding++;
                             break;
                         case ShaderResourceKind.Texture2DArray:
-                            WriteTexture2DArray(sb, rd, textureBinding);
+                            if (!resourcesUsed.Contains(rd))
```

Comment:
Is this supposed to be a negative condition?

Reply:
No -.-